src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -709,6 +709,58 @@ static ucs_status_t uct_tcp_sockcm_ep_client_init(uct_tcp_sockcm_ep_t *cep,
     return status;
 }
 
+static ucs_status_t uct_tcp_sockcm_ep_server_check_cm(const uct_ep_params_t *params,
+                                                      uct_tcp_sockcm_ep_t *tcp_ep)
```

Comment:
according to what this func does it should be like
```
static ucs_status_t uct_tcp_sockcm_ep_set_cm(uct_tcp_sockcm_ep_t *tcp_ep,
                                                       const uct_tcp_sockcm_t *params_tcp_sockcm)
```

Reply:
@evgeny-leksikov why? this way i would have to retrieve the param_tcp_sockcm from outside this function (or do it in its args).
the way that it is now is that params are passed to the function and all the related work to cms is done inside it.
also, if the cms are the same, then the cm of the ep stays the same it was and this function doesn't do anything, so it doesn't set a cm (as the name you suggested implies).

Reply:
`check/test/_is_` functions should not have side effects, they just returns true/false/status.
> the way that it is now is that params are passed to the function and all the related work to cms is done inside it.

CM flag field has to be checked earlier and must be present here, so you can pass CM unconditionally:
```
status = uct_tcp_sockcm_ep_set_cm(ep, params->cm);
```
this makes code simple inside and outside of the function.

> if the cms are the same, then the cm of the ep stays the same it was and this function doesn't do anything, so it doesn't set a cm

OK, rename "set->update" but IMO if CMs are the same result is the same, passed CM is set :)

Reply:
@evgeny-leksikov `uct_cm_use_ep_params` is doing the check for the `UCT_EP_PARAM_FIELD_CM` flag so i moved its call to before calling this function. 
since this function is working only on the cm from 'params', i changed this function to accept only 'params->cm' and not 'params'
i renamed to uct_tcp_sockcm_ep_server_update_cm, it sounds better to me. 'update' implies that a cm is already set (which is true) and might just need to update it.